### ISSUE TITLE
Tweaking completion text based on symbols after the cursor

### ISSUE THREAD

**Issue by kirillbobyrev** [clangd/clangd#387]
There is a frequent pattern of the completion text interfering with the existing source code after the cursor.

1. I'd like to change the method call to a different one: `Foo.bar()` -> `Foo->baz()`. This happens very often when I copy-paste a couple of lines and would like to access different method. The initial code I have is `Foo.bar()`.
2. What I typically do is put my cursor onto the beginning of `bar()` and call `dw:`: `Foo.[[cursor]]bar()` -> `Foo.[[cursor]]()`.
3. Then I start typing the method I'd like to call and as soon as it appears I immediately select it. The result is `Foo.baz[[cursor]]()()`. Now I have to press `d2w` again.

In Vim I could do `c%` as suggested by @sam-mccall, but it's hard to remember and I'm not sure many editors have convenient hotkeys for the same motion. It'd be to look a couple of characters ahead of the cursor and modify the completion item so that there is no additional pair of parenthesis after the inserted text.

There might be other cases where this makes sense. Examples:

* Completing in the middle of identifier and replacing the rest of it with the new text: e.g. `vari[[cursor]]able` -> `various[[cursor]]` (right now it would be `variousable`).

Do you know any other cases with similar behaviour?

**Comment by HighCommander4** [comment]
The "double parentheses" issue is on file as #119.

**Comment by kirillbobyrev** [comment]
@HighCommander4 Good catch, thanks! I'll close that one as I'd like the current issue to be a "meta-issue" one for more similar cases.

**Comment by JVApen** [comment]
As requested, posting this here:
I've seen variations of this occuring when trying to use the autocomplete to add the required includes.
````
auto f()
{
    return g^(1, 2);
}
````
Became
````
#include "g.h"
auto f()
{
    return g()(1, 2);
}
````
And
````
auto f()
{
    return g^<int, int>();
}
````
Became
````
#include "g.h"
auto f()
{
    return g<T1, T2><int, int>();
}
````

**Comment by timon-ul** [comment]
This is being implemented by https://github.com/llvm/llvm-project/pull/197249

**Comment by HighCommander4** [comment]
Closing as fixed by https://github.com/llvm/llvm-project/pull/197249.
